.github/workflows/test-upstream.yaml

```diff
@@ -61,29 +64,40 @@ jobs:
           path: linkml/.venv
           key: venv-${{ matrix.python-version }}-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
 
-      # make extra sure we're removing any old version of linkml-runtime that exists
-      - name: uninstall potentially cached linkml-runtime
-        working-directory: linkml
-        run: poetry run pip uninstall linkml-runtime
-
-      # we are not using linkml-runtime's lockfile, but simulating what will happen
-      # when we merge this and update linkml's lockfile
-      - name: add linkml-runtime to lockfile
-        working-directory: linkml
-        run: poetry add ../linkml-runtime
-
       # use correct pydantic version
       - name: install pydantic
         working-directory: linkml
         run: poetry add pydantic@^${{ matrix.pydantic-version }}
 
+      - name: remove potentially cached linkml-runtime version
+        working-directory: linkml
+        run: poetry run pip uninstall linkml-runtime -y
+
+      - name: get linkml-runtime short hash
+        working-directory: linkml-runtime
+        run: |
+          LINKML_RUNTIME_COMMIT=$(git rev-parse --short HEAD)
+          echo "LINKML_RUNTIME_COMMIT=$LINKML_RUNTIME_COMMIT" >> "$GITHUB_ENV"
```

Comment:
I think this step can be replaced with running this in the `linkml-runtime` directory:

```
poetry self add "poetry-dynamic-versioning[plugin]"
poetry dynamic-versioning
```

Using the dynamic versioning plugin via the CLI will modify the `version` field in `pyproject.toml` according to the current Git status and leave the change in place. 

Then in the next step you should be able to just do the `poetry add ../linkml-runtime`. I think 😂. If that works that feels a little better to me than using the fake `1.99.0+x` version.

Reply:
(agreed it's ugly and i hate it lol)